cpp/src/c_api/replicate_edgelist.cpp

```diff
@@ -131,27 +187,84 @@ extern "C" cugraph_error_code_t cugraph_replicate_edgelist(
   auto p_weights =
     reinterpret_cast<cugraph::c_api::cugraph_type_erased_device_array_view_t const*>(weights);
 
-  /*
-  reinterpret_cast<cugraph::c_api::cugraph_induced_subgraph_result_t**>(result),
-  reinterpret_cast<cugraph::c_api::cugraph_error_t**>(error)
-  */
-
-  // FIXME: Support int64_t and float as well
-  return cugraph_replicate_edgelist<int32_t, float>(
-    *p_handle->handle_,
-    p_src,
-    p_dst,
-    p_weights,
-    reinterpret_cast<cugraph::c_api::cugraph_induced_subgraph_result_t**>(result),
-    reinterpret_cast<cugraph::c_api::cugraph_error_t**>(error));
-
-  /*
-  return dummy_function<int32_t, float>(
-   *p_handle->handle_,
-   p_src,
-   p_dst,
-   p_weights,
-   reinterpret_cast<cugraph::c_api::cugraph_induced_subgraph_result_t**>(result),
-   reinterpret_cast<cugraph::c_api::cugraph_error_t**>(error));
-   */
+  auto p_edge_ids =
+    reinterpret_cast<cugraph::c_api::cugraph_type_erased_device_array_view_t const*>(edge_ids);
+
+  auto p_edge_type_ids =
+    reinterpret_cast<cugraph::c_api::cugraph_type_erased_device_array_view_t const*>(edge_type_ids);
+
+  CAPI_EXPECTS(p_src->size_ == p_dst->size_,
+               CUGRAPH_INVALID_INPUT,
+               "Invalid input arguments: src size != dst size.",
+               *error);
+  CAPI_EXPECTS(p_src->type_ == p_dst->type_,
+               CUGRAPH_INVALID_INPUT,
+               "Invalid input arguments: src type != dst type.",
+               *error);
+
+  CAPI_EXPECTS((weights == nullptr) || (p_weights->size_ == p_src->size_),
+               CUGRAPH_INVALID_INPUT,
+               "Invalid input arguments: src size != weights size.",
+               *error);
+
+  cugraph_data_type_id_t edge_type;
+  cugraph_data_type_id_t weight_type;
+
+  if (p_src->size_ < int32_threshold) {
+    edge_type = p_src->type_;
+  } else {
+    edge_type = cugraph_data_type_id_t::INT64;
+  }
+
+  if (weights != nullptr) {
+    weight_type = p_weights->type_;
+  } else {
+    weight_type = cugraph_data_type_id_t::FLOAT32;
+  }
+
+  cugraph_data_type_id_t edge_type_id_type = cugraph_data_type_id_t::INT32;
+  if (edge_type_ids != nullptr) { edge_type_id_type = p_edge_type_ids->type_; }
+
+  CAPI_EXPECTS((edge_ids == nullptr) || (p_edge_ids->type_ == edge_type),
```

Comment:
If you make the above recommended change this is unnecessary.

Reply:
Isn't this code doing what you recommended above but for `edge_type_ids`?

Reply:
I think what you've created is correct.  This comment is kind of obsolete at this point.  This was targeted at the error check `p_edge_ids->type_ == edge_type`, which is obsolete... but not in your code anymore.